openedx/core/djangoapps/catalog/utils.py

```diff
@@ -389,13 +389,17 @@ def get_fulfillable_course_runs_for_entitlement(entitlement, course_runs):
     search_time = datetime.datetime.now(UTC)
     for course_run in course_runs:
         course_id = CourseKey.from_string(course_run.get('key'))
-        is_enrolled = CourseEnrollment.is_enrolled(entitlement.user, course_id)
+        (user_enrollment_mode, is_active) = CourseEnrollment.enrollment_mode_for_user(
+            user=entitlement.user,
+            course_id=course_id
+        )
+        is_enrolled_in_mode = user_enrollment_mode == entitlement.mode
         if is_course_run_entitlement_fulfillable(course_id, entitlement, search_time):
-            if (is_enrolled and
+            if (is_enrolled_in_mode and
```

Comment:
Yes, because it allow the current session to still be displayed in the change session dropdown. It will then be marked as (current session).
<img width="1027" alt="screen shot 2018-03-08 at 2 40 05 pm" src="https://user-images.githubusercontent.com/2854941/37172398-a66c02e6-22de-11e8-98fb-700048d59b4d.png">

Reply:
This logic does not change in this PR and is from the original change a month or so ago.

Reply:
why was the card rendering without the View Course button? because it was unexpired but had no enrollable runs?